pinot-common/src/main/java/com/linkedin/pinot/common/utils/webhdfs/WebHdfsV1Client.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.utils.webhdfs;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpMethod;
+import org.apache.commons.httpclient.methods.FileRequestEntity;
+import org.apache.commons.httpclient.methods.PutMethod;
+import org.apache.commons.httpclient.methods.RequestEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class WebHdfsV1Client {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebHdfsV1Client.class);
+  private static final String CREATE = "CREATE";
+  private static final String LOCATION = "Location";
+  private static final boolean DEFAULT_OVERWRITE = true;
+  private static final int DEFAULT_PERMISSION = 755;
+
+  //Web hdfs path template has 4 parameters: host/port/hdfs_path/operation
+  public static String WEB_HDFS_PATH_TMEPLATE = "http://%s:%s/webhdfs/v1%s?op=%s&overwrite=%s&permission=%s";
```

Comment:
Typo in template.

Is it possible that this can be https too? Might be better to have a single string for the API endpoint (eg. "http://foo:1234/webhdfs/v1"), that way people can specify the proto, host and port at the same time.